src/app/applicationinstancemanager.cpp

```diff
@@ -35,11 +35,12 @@
 #include <QDebug>
 #include <QSharedMemory>
 
+#include "base/profile.h"
 #include "qtlocalpeer/qtlocalpeer.h"
 
 ApplicationInstanceManager::ApplicationInstanceManager(const QString &appId, QObject *parent)
     : QObject {parent}
-    , m_peer {new QtLocalPeer {this, appId}}
+    , m_peer {new QtLocalPeer {Profile::instance()->location(SpecialFolder::Config), this}}
```

Comment:
I wouldn't create this additional dependency. Just accept `instance path` as constructor parameter. Also `appId` can be calculated in this constructor (based on `instance path` parameter) instead of obtaining it from outside.

Reply:
Not sure I understand you, anyway I pushed an updated version.

Reply:
>Just accept instance path as constructor parameter. 

If we want `ApplicationInstanceManager` to handle app instances, then perhaps it shouldn't take any "app instance" related input parameters. The class should decide app ID on its own. Am I right?

Reply:
Also reasonable... Well, then it is still acceptable to get the necessary information from other application components (e.g. Profile) inside this class.

Reply:
But I still wouldn't add unnecessary dependencies. There is nothing wrong with some component (`ApplicationInstanceManager`) receiving input from a higher-level component (`Application`). Even if it is called a `manager`, this does not detract from its managerial functions. But it simplifies the dependency of application components. For example, you can clearly see from the interface of `ApplicationInstanceManager` class that before creating an instance of it, you need to get the path to the application configuration, i.e. the `Profile` instance must be created before that.
But if you really dislike to call it a manager in this case you can suggest more appropriate name.

Reply:
>But if you really dislike to call it a manager in this case you can suggest more appropriate name.

I would choose `ApplicationInstance` and also defer the renaming to another PR.

Reply:
>defer the renaming to another PR.

👍